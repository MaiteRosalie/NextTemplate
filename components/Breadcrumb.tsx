import React, { useState, useRef, useEffect } from 'react';
import styled from '@emotion/styled';
import { Box, Image, Flex, Text } from 'rebass';

const Wrapper = styled(Flex)`
  position: relative;
  overflow: hidden;
`;

const Background = styled(Box)`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
  z-index: -1;
  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    filter: brightness(0.5);
  }
`;
export type BreadCrumbProps = {
  src: string;
  title: string;
};

export const BreadCrumb = ({ title, src }: BreadCrumbProps) => {
  const [animated, setAnimated] = useState(false);
  const imageRef = useRef<HTMLImageElement>(null);

  const onLoad = () => {
    setAnimated(true);
  };

  useEffect(() => {
    if (imageRef.current && imageRef.current.complete) {
      setAnimated(true);
    }
  }, []);
  return (
    <Wrapper
      as="header"
      alignItems="center"
      justifyContent="center"
      p={3}
      pt="5rem"
      minHeight={['40vh']}
    >
      <Background as="picture">
        <source srcSet={`./static/images/${src}-1500.png`} media="(min-width: 800px)" />
        <source srcSet={`./static/images/${src}-800.png`} media="(max-width: 800px)" />
        <Image
          src={`./static/images/${src}-1500.png`}
          alt={title}
          onLoad={onLoad}
          ref={imageRef}
          css={{
            transition: `opacity 300ms cubic-bezier(0.4, 0, 0.2, 1),
            transform 300ms cubic-bezier(0.4, 0, 0.2, 1)`,
            opacity: animated ? '1' : '0',
          }}
        />
      </Background>
      <Flex
        width={1}
        flexDirection="column"
        alignItems="center"
        justifyContent="center"
        css={{
          transition: `opacity 600ms cubic-bezier(0.4, 0, 0.2, 1) 150ms,
          transform 600ms cubic-bezier(0.4, 0, 0.2, 1) 150ms`,
          opacity: animated ? '1' : '0',
          transform: animated ? 'translateY(-1rem)' : 'translateY(0rem)',
        }}
      >
        <Text as="h1" color="white" fontSize={[5, 6]} fontFamily="heading" textAlign="center">
          {title}
        </Text>
      </Flex>
    </Wrapper>
  );
};
