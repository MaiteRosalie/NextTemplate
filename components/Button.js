import React from 'react';
import { Button as BaseButton } from 'rebass';

export const Button = ({ disabled, children, rounded, ...props }) => {
  let tag;
  tag = props.href ? 'a' : 'button';
  tag = tag === 'a' && disabled ? 'span' : tag;

  return (
    <BaseButton
      sx={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: rounded ? '30px' : '4px',
      }}
      as={tag}
      disabled={disabled && tag === 'button' ? disabled : undefined}
      {...props}
    >
      {children}
    </BaseButton>
  );
};
