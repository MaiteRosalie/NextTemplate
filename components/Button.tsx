import React from 'react';
import { Button as BaseButton, ButtonProps as BaseButtonProps } from 'rebass';

type ButtonProps = Pick<
  BaseButtonProps,
  Exclude<keyof BaseButtonProps, 'css' | 'as' | 'variant' | 'disabled'>
> & {
  rounded?: boolean;
};

export const Button = ({ children, rounded, ...props }: ButtonProps) => {
  return (
    <BaseButton
      sx={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: rounded ? '30px' : '2px',
      }}
      as={props.href ? 'a' : 'button'}
      {...props}
    >
      {children}
    </BaseButton>
  );
};
