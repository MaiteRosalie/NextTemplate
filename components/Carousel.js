import React from 'react';
import Carousel from 'react-multi-carousel';
import styled from '@emotion/styled';

const StyledCarousel = styled(Carousel)`
  overflow: hidden;
  position: relative;
  .react-multi-carousel-track {
    list-style: none;
    padding: 0;
    margin: 0;
    display: flex;
    position: relative;
    will-change: transform, transition;
    height: 100%;
    width: ${({ width }) => width}%;
  }

  .react-multiple-carousel__arrow {
    position: absolute;
    outline: 0;
    transition: all 0.5s;
    border-radius: 50%;
    z-index: 1000;
    border: 0;
    width: 2.3rem;
    height: 2.3rem;
    background-color: transparent;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .react-multiple-carousel__arrow::before {
    font-size: 1.3rem;
    display: block;
    z-index: 2;
    position: relative;
  }

  .react-multiple-carousel__arrow--left {
    left: calc(4% + 1px);
  }

  .react-multiple-carousel__arrow--left::before {
    content: '◀';
  }

  .react-multiple-carousel__arrow--right {
    right: calc(4% + 1px);
  }

  .react-multiple-carousel__arrow--right::before {
    content: '▶';
  }

  .react-multi-carousel-dot-list {
    display: flex;
    padding: 0;
    margin: 0;
    list-style: none;
    text-align: center;
    justify-content: flex-start;
  }

  .react-multi-carousel-dot button {
    transition: opacity 0.5s;
    width: 10px;
    height: 10px;
    border-radius: 50%;
    margin: 0 8px;
    padding: 0;
    opacity: 0.5;
    border: none;
    background: white;
    cursor: pointer;
  }

  .react-multi-carousel-dot--active button {
    opacity: 1;
  }

  .react-multi-carousel-item {
    transform-style: preserve-3d;
    backface-visibility: hidden;
  }

  @media all and (-ms-high-contrast: none), (-ms-high-contrast: active) {
    .react-multi-carousel-item {
      flex-shrink: 0 !important;
    }

    .react-multi-carousel-track {
      overflow: visible !important;
    }
  }
`;
const Component = props => <StyledCarousel {...props} width={props.children.length * 100} />;
export { Component as Carousel };
