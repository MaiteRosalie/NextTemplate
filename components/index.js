export * from "./Layout";
export * from "./Modal";
export * from "./Button";
export * from "./Navigation";
export * from "./FancyText";
