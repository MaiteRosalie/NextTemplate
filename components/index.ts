export * from './Layout';
export * from './Modal';
export * from './ModalContext';
export * from './Button';
export * from './Navigation';
export * from './FancyText';
export * from './Navigation';
export * from './Carousel';
export * from './Header';
export * from './Features';
export * from './Steps';
export * from './CTA';
export * from './Footer';
export * from './Testimonials';
