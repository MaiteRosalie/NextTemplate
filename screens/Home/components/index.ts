export * from './Header';
export * from './Features';
export * from './Steps';
export * from './CTA';
export * from './Testimonials';
export * from './Icons';
