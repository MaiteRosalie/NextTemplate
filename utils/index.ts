export * from './tests';
export * from './devices';
export * from './location';
