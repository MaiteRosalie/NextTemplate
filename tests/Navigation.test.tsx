import React from 'react';
import { fireEvent } from '@testing-library/react';

import { render } from '../utils/tests';
import { Navigation } from '../components';

const setup = () => {
  const toggleHandler = jest.fn();
  const utils = render(<Navigation toggle={toggleHandler} />);
  return {
    ...utils,
    toggleHandler,
  };
};

describe('rendering', () => {
  test('renders as expected', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});

describe('callbacks', () => {
  test('executes toggle handler once when menu icon is clicked', () => {
    const { container, toggleHandler } = setup();
    const button = container.querySelector('.mb-open-btn') as HTMLDivElement;

    fireEvent.click(button);
    expect(toggleHandler).toBeCalledTimes(1);
  });
});
