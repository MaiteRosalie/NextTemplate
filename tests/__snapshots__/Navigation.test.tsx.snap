// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<div>
  <div
    class="show-md mb-open-btn css-l6kea7-Open esgq4xm3"
  >
    <div
      class="css-1s4dz1f"
    />
    <div
      class="css-12ptazb"
    />
    <div
      class="css-1s4dz1f"
    />
  </div>
  <nav
    class="css-1dcz2ie-Nav esgq4xm1"
  >
    <svg
      class="show-md css-un36pl-Close esgq4xm2"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M278.6 256l68.2-68.2c6.2-6.2 6.2-16.4 0-22.6-6.2-6.2-16.4-6.2-22.6 0L256 233.4l-68.2-68.2c-6.2-6.2-16.4-6.2-22.6 0-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3l68.2 68.2-68.2 68.2c-3.1 3.1-4.7 7.2-4.7 11.3 0 4.1 1.6 8.2 4.7 11.3 6.2 6.2 16.4 6.2 22.6 0l68.2-68.2 68.2 68.2c6.2 6.2 16.4 6.2 22.6 0 6.2-6.2 6.2-16.4 0-22.6L278.6 256z"
      />
    </svg>
    <div
      class="css-4cffwv"
    >
      <a
        class="esgq4xm0 css-1m8yfox-HL"
        href="./"
      >
        About
      </a>
      <a
        class="esgq4xm0 css-1m8yfox-HL"
        href="./"
      >
         What is MCA?
      </a>
      <a
        class="esgq4xm0 css-1m8yfox-HL"
        href="./"
      >
        Apply
      </a>
      <a
        class="esgq4xm0 css-1m8yfox-HL"
        href="./"
      >
        Contact
      </a>
    </div>
    <div
      class="css-4cffwv"
    >
      <a
        class="esgq4xm0 css-1m8yfox-HL"
        href="tel:1(877)498-4344"
      >
        Call Now 1 (877) 498-4344
      </a>
      <a
        class="show-md esgq4xm0 css-1m8yfox-HL"
        href="tel:1(877)498-4344"
      >
        Login
      </a>
      <a
        class="hide-md css-167ay55"
        href="tel:1(877)498-4344"
      >
        Login
      </a>
    </div>
  </nav>
</div>
`;
